themes/src/main/resources/theme/base/login/resources/js/utils.js

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *  * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ *  * and other contributors as indicated by the @author tags.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+const formatNumber = (input, format) => {
+    if (!input) {
+        return "";
+    }
+
+    // array holding the patterns for the number of expected digits in each part
+    const digitPattern = format.match(/{\d+}/g);
+
+    if (!digitPattern) {
+        return "";
+    }
+
+    let maxSize = 0;
+
+    // calculate the maximum size of the given pattern based on the sum of the expected digits
+    for (const p of digitPattern) {
+        maxSize += parseInt(p.replace("{", "").replace("}", ""));
+    }
+
+    // keep only digits
+    let rawValue = input.replace(/\D+/g, '');
+
+    // make sure the value is a number
+    if (parseInt(rawValue) != rawValue) {
+        return "";
+    }
+
+    // make sure the number of digits do not exceed the maximum size
+    if (rawValue.length > maxSize) {
+        rawValue = rawValue.substring(0, maxSize);
+    }
+
+    console.log(rawValue);
+
+    let formatter = "^";
+
+    // build the regex based based on the expected digits in each part
+    for (const p of digitPattern) {
+        formatter += "(\\d" + p + ")";
+    }
+
+    // if the current digits match the pattern we have each group of digits in an array
+    let digits = new RegExp(formatter).exec(rawValue);
+
+    // no match, return the raw value without any format
+    if (!digits) {
+        return input;
+    }
+
+    let result = format;
+
+    // finally format the current digits accordingly to the given format
+    for (let i = 0; i < digitPattern.length; i++) {
+        result = result.replace(digitPattern[i], digits[i + 1]);
+    }
+
+    return result;
+}
+
+document.querySelectorAll('[data-kc-number-format]')
+    .forEach(input => {
+        input.onkeyup = () => input.value = formatNumber(input.value, input.getAttribute('data-kc-number-format'));
+        document.forms[0].addEventListener('submit', (e) => {
+            let rawFormat = input.getAttribute('data-kc-number-raw-format');
```

Comment:
Can be a `const`, as it is never re-assigned.
```suggestion
            const rawFormat = input.getAttribute('data-kc-number-raw-format');
```